docs/architecture/index.md

```diff
@@ -0,0 +1,9 @@
+# Architecture
+
+This living document serves as a high-level documentation to explain the conceptual choices and tool choices in the Gutenberg repository.
+
+- [Modularity and WordPress Packages](docs/architecture/modularity.md).
+- [Block Editor Performance](docs/architecture/performance.md).
+- What are the decision decisions behind the Data Module?
+- [Why is Puppeteer the tool of choice for end-to-end tests?](/docs/architecture/automated-testing.md)
+- [What's the difference between the different editor packages? What's the purpose of each package?](/docs/architecture/modularity.md#whats-the-difference-between-the-different-editor-packages-whats-the-purpose-of-each-package)
```

Comment:
In 25d906ed63, I added a section to the existing "Modularity" document to outline the purpose of the editor packages.